internal/cloud/keyvault.go

```diff
@@ -55,134 +53,103 @@ func getAzureClientAuthMethod() clientAuthMethod {
 	return mode
 }
 
-// NewAzureClient returns a new Azure Key Vault client
-// By default, the authorizer will be from Managed Identity
-// If user sets AKV_AUTH_METHOD to AKV_AUTH_FROM_CLI, it will use Azure CLI instead
-func NewAzureClient() (*keyvault.BaseClient, error) {
-	var (
-		authorizer autorest.Authorizer
-		err        error
-	)
+// AzureCredential returns an Azure TokenCredential
+func AzureCredential() (credential azcore.TokenCredential, err error) {
 	authMethod := getAzureClientAuthMethod()
 	switch authMethod {
 	case authorizerFromMI:
-		authorizer, err = auth.NewAuthorizerFromEnvironment()
+		credential, err = azidentity.NewManagedIdentityCredential(nil)
 		if err != nil {
-			return nil, fmt.Errorf("%s, origin error: %w", errMsgAuthorizerFromMI, err)
+			err = fmt.Errorf("%s. Original error: %w", errMsgAuthorizerFromMI, err)
 		}
 	case authorizerFromCLI:
-		authorizer, err = auth.NewAuthorizerFromCLI()
+		credential, err = azidentity.NewAzureCLICredential(nil)
 		if err != nil {
-			return nil, fmt.Errorf("%s, origin error: %w", errMsgAuthorizerFromCLI, err)
+			err = fmt.Errorf("%s. Original error: %w", errMsgAuthorizerFromCLI, err)
 		}
 	default:
-		return nil, errors.New(errMsgUnknownAuthorizer)
+		err = errors.New(errMsgUnknownAuthorizer)
 	}
-	client := keyvault.New()
-	client.Authorizer = authorizer
-	return &client, nil
+	return
 }
 
-// Key represents a remote key in the Azure Key Vault.
-type Key struct {
-	Client *keyvault.BaseClient
+// KeyVault represents a Azure KeyVault Vault.
+type KeyVault struct {
```

Comment:
According to the [AKV doc](https://learn.microsoft.com/azure/key-vault/certificates/about-certificates#composition-of-a-certificate), 
![cert](https://learn.microsoft.com/azure/key-vault/media/azure-key-vault.png)
this structure should be called
```go
type Certificate struct {
```
To make the reference meaning, it is recommended to rename the package from `cloud` to `keyvault` so that we can access the struct as `keyvault.Certificate`.